eclass/python-utils-r1.eclass

```diff
@@ -1012,38 +1027,63 @@ python_wrapper_setup() {
 		# note: we don't use symlinks because python likes to do some
 		# symlink reading magic that breaks stuff
 		# https://bugs.gentoo.org/show_bug.cgi?id=555752
-		cat > "${workdir}/bin/python" <<-_EOF_ || die
-			#!/bin/sh
-			exec "${PYTHON}" "\${@}"
-		_EOF_
-		cp "${workdir}/bin/python" "${workdir}/bin/python${pyver}" || die
-		chmod +x "${workdir}/bin/python" "${workdir}/bin/python${pyver}" || die
+		echo '#!/bin/sh' > "${workdir}/bin/python" || die
 
 		local nonsupp=( "python${pyother}" "python${pyother}-config" )
 
 		# CPython-specific
 		if [[ ${EPYTHON} == python* ]]; then
+			local sysrootlib=${SYSROOT-${ROOT%/}}${EPREFIX}/usr/$(get_libdir)
+
 			cat > "${workdir}/bin/python-config" <<-_EOF_ || die
 				#!/bin/sh
-				exec "${PYTHON}-config" "\${@}"
+				exec "${SYSROOT-${ROOT%/}}${EPREFIX}/usr/bin/${EPYTHON}-config" "\${@}"
 			_EOF_
 			cp "${workdir}/bin/python-config" \
 				"${workdir}/bin/python${pyver}-config" || die
 			chmod +x "${workdir}/bin/python-config" \
 				"${workdir}/bin/python${pyver}-config" || die
 
+			# Use host-specific data from SYSROOT. Python 2 looks for
+			# _sysconfigdata while 3 uses _PYTHON_SYSCONFIGDATA_NAME.
+			ln -s "${sysrootlib}/${EPYTHON}"/_sysconfigdata*.py "${workdir}"/lib/_sysconfigdata.py || die
+
+			# Use distutils/sysconfig from SYSROOT as parts of it have
+			# GENTOO_LIBDIR baked in at Python build time.
```

Comment:
is GENTOO_LIBDIR the only issue here?

Reply:
For those particular files, yes, I have just diff'd them to check. Does that mean we can avoid the `PYTHONPATH` issue? Not quite. You still have to override the file pointed at by `_PYTHON_SYSCONFIGDATA_NAME`. This variable cannot be a full path. It also needs to be set via a wrapper because as soon as you set it within the Portage environment, Portage itself keels over. Maybe we could patch in support for a full path but I'm not sure how we'd prevent it from screwing with Portage.

I need to rebase this because your patch was reverted in the end.

Reply:
Ok, so if we got rid of GENTOO_LIBDIR altogether will it reduce the needed patch or not at all?

Reply:
Only the following call to ln.